src/app/tabs/tabs.component.ts

```diff
@@ -63,9 +65,9 @@ export class TabsComponent extends RouterEnter {
   }
 
   private _initialise() {
-    this.showChat = false;
-    this.showReview = false;
-    this.showEvents = false;
+    this.showChat = this.showChat || false;
+    this.showReview = this.showReview || false;
+    this.showEvents = this.showEvents || false;
```

Comment:
If we change it to this, we may have the following problem:
1. user 1 login and should see chat/review/events, he will see them
2. user 1 logout
3. user 2 login and shouldn't see chat/review/events, but he will see them at first because those 3 variables are true

Reply:
oh! i noticed there is a ugly UX problem:
switching between pages will cause tabs keep getting hide and show.

the chance that a 2nd user login and see the tab getting hidden once is still a better solution than keep showing the "UX problem" to every user. What do you think @shawnm0705 ?

Reply:
@trtshen in that case, we need to store a variable in local storage to indicate if we should display those in tab. And remember to clear it when user logout

Reply:
@shawnm0705 no need, i've tested this 3 lines changes, works well enough for fixing "jumpy tabs". And, these values can be safely "refreshed" at every page refresh.